lldb/source/Plugins/ExpressionParser/Swift/SwiftExpressionParser.cpp

```diff
@@ -1471,6 +1585,148 @@ bool SwiftExpressionParser::Complete(CompletionRequest &request, unsigned line,
+  if (!func_decl) {
+    log->Printf("[SetCallFromSinkToGenericFunction] Could not set the call: no "
+                "func decl.");
+    return false;
+  }
+
+  auto *generic_func_decl = manipulator.GetGenericDecl();
+  if (!generic_func_decl) {
+    log->Printf("[SetCallFromSinkToGenericFunction] Could not set the call: no "
+                "generic func decl.");
+    return false;
+  }
+
+  auto *sink_decl = manipulator.GetSinkDecl();
+  if (!sink_decl) {
+    log->Printf("[SetCallFromSinkToGenericFunction] Could not set the call: no "
+                "sink decl.");
+    return false;
+  }
+
+  auto expr_func_name = mangler.mangleEntity(wrapper_decl);
+  auto wrapped_func_name = mangler.mangleEntity(func_decl);
+  auto generic_func_name = mangler.mangleEntity(generic_func_decl);
+  auto sink_func_name = mangler.mangleEntity(sink_decl);
+
+  llvm::Function *lldb_expr_func = module.getFunction(expr_func_name);
+  llvm::Function *wrapped_func = module.getFunction(wrapped_func_name);
+  llvm::Function *generic_func = module.getFunction(generic_func_name);
+  llvm::Function *sink_func = module.getFunction(sink_func_name);
+
+  assert(lldb_expr_func && wrapped_func && generic_func && sink_decl);
+  if (!lldb_expr_func || !wrapped_func || !generic_func || !sink_func) {
+    log->Printf("[SetCallFromSinkToGenericFunction] Could not set the call: "
+                "could not find one of the required functions in the IR.");
+    return false;
+  }
+
+  auto &basic_blocks = lldb_expr_func->getBasicBlockList();
+  // The wrapper function should only have one basic block whith materialization
+  // instructions and the call to the sink.
+  if (basic_blocks.size() != 1) {
+    log->Printf("[SetCallFromSinkToGenericFunction] Could not set the call: "
+                "wrapper function has %zu basic blocks.",
+                basic_blocks.size());
+    return false;
+  }
+
+  auto &basic_block = basic_blocks.back();
+  if (basic_block.getInstList().size() == 0) {
+    log->Printf("[SetCallFromSinkToGenericFunction] Could not set the call: "
+                "basic block has no instructions.");
+    return false;
+  }
+
+  auto &it = basic_block.getInstList().back();
+  // Initialize the builder from the last instruction since we want to place the
+  // new call there.
+  llvm::IRBuilder<> builder(&it);
+
+  auto *generic_func_type = generic_func->getFunctionType();
+  auto num_params = generic_func_type->getNumParams();
+  // There should be 3 params, the raw pointer, the self type, and the pointer
+  // to metadata
+  if (num_params != 3) {
+    log->Printf("[SetCallFromSinkToGenericFunction] Could not set the call: "
+                "generic function has %u parameters",
+                num_params);
+    return false;
+  }
+
+  llvm::Type *param1 = generic_func_type->getParamType(1);
+  llvm::Type *param2 = generic_func_type->getParamType(2);
+
+  // Find the call to the sink.
+  llvm::CallInst *sink_call = nullptr;
+  for (auto &I : basic_block) {
```

Comment:
`basic_block.instructionsWithoutDebug(true)`

The debug info intrinsics are also calls, so this will be slightly less confusing.